src/libs/ReportUtils.js

```diff
@@ -3135,6 +3198,41 @@ function getTaskAssigneeChatOnyxData(accountID, assigneeEmail, assigneeAccountID
     };
 }
 
+/**
+ * Returns the number of receipts associated with an IOU report that are still being scanned.
+ * Note that we search the IOU report for this number, since scanning receipts will be whispers
+ * in the IOU report for only the submitter and we can get an accurate count.
+ *
+ * @param {Object|null} iouReport
+ * @returns {Number}
+ */
+function getNumberOfScanningReceipts(iouReport) {
+    const reportID = lodashGet(iouReport, 'reportID');
+    const reportActions = ReportActionsUtils.getAllReportActions(reportID);
+    return _.reduce(
+        reportActions,
+        (count, reportAction) => {
+            if (!ReportActionsUtils.isMoneyRequestAction(reportAction)) {
+                return count;
+            }
+            const transaction = TransactionUtils.getTransaction(reportAction.originalMessage.IOUTransactionID);
+            return count + Number(TransactionUtils.hasReceipt(transaction) && TransactionUtils.isReceiptBeingScanned(transaction));
+        },
+        0,
+    );
+}
+
+/**
+ * Get the last 3 transactions with receipts of an IOU report that will be displayed on the report preview
+ *
+ * @param {Object} reportPreviewAction
+ * @returns {Object}
+ */
+function getReportPreviewDisplayTransactions(reportPreviewAction) {
+    const transactionIDs = lodashGet(reportPreviewAction, ['childLastReceiptTransactionIDs'], '').split(',');
+    return _.map(transactionIDs, (transactionID) => TransactionUtils.getTransaction(transactionID));
```

Comment:
This may cause a crash if the transaction is not there which will return `{}` and we will try to access `receipt.state` on that. Let's use the `hasReceipt` to be safe